SwiftyStoreKit/AppleReceiptValidator.swift

```diff
@@ -110,9 +110,15 @@ public struct AppleReceiptValidator: ReceiptValidator {
 					sandboxValidator.validate(receiptData: receiptData, completion: completion)
 				} else {
 					if receiptStatus.isValid {
-						completion(.success(receipt: receiptInfo))
+                        if case .sandbox = self.service {
+                            var mutableReceiptInfo = receiptInfo
+                            mutableReceiptInfo["is_sandbox"] = "1" as AnyObject
```

Comment:
I am hesitant to start adding additional information to the receipt, especially as some apps may not expect this data and it adds a layer of complexity for developers.

Instead, a better approach might be to make the `service` property public with a `private(set)`.

Even so, the comment above this clearly indicates you can check the provided receipt's status code for `21007`, to see if the app is being tested in the sandbox (esp. during app review).

Reply:
It is mainly used for statistics. I think only make `service` get public will not work, should use the same `AppleReceiptValidator` instance and change service to `sandbox` and `validate()` again, or add current used `AppleReceiptValidator` to the completion closure